.travis.yml

```diff
@@ -22,33 +26,64 @@ matrix:
     - os: linux
       language: python
       python: "2.7"
+      env:
+        TEST_COMPILER=true
     - os: linux
       language: python
       python: "3.5"
       env:
+        TEST_COMPILER=true
         TEST_REGISTRY=true
       services:
         - postgresql
     - os: linux
       language: python
       python: "3.6"
       env:
+        TEST_COMPILER=true
         TEST_REGISTRY=true
       services:
         - postgresql
+    - os: linux
+      language: node_js
+      cache:
+        directories:
+          - $HOME/.npm
+      node_js:
+        - node
+        - lts
+      env:
+        TEST_CATALOG=true
 
 install:
-  # install quilt from setup.py
-  - travis_retry pip install ./compiler['tests']
+  - |
+    # install quilt from setup.py
+    if [[ "$TEST_COMPILER" = true ]]; then
+      travis_retry pip install ./compiler['tests']
+    fi
   - |
     if [[ "$TEST_REGISTRY" = true ]]; then
       travis_retry pip install -r registry/requirements.txt
       pip install registry
     fi
+  - |
+    if [[ "$TEST_CATALOG" = true ]]; then
+      cd catalog && npm ci
+    fi
 
 script:
-  - pytest compiler
+  - |
+    if [[ "$TEST_COMPILER" = true ]]; then
+      pytest --cov=./ compiler
```

Comment:
should this be `./compiler`? the registry case uses `./registry`

Reply:
idk, it was `- pytest compiler` for quite some time (see diff and history for details), and i haven't touched it

Reply:
OK sounds good